src/test/java/org/broadinstitute/hellbender/tools/walkers/vqsr/scalable/ScoreVariantAnnotationsIntegrationTest.java

```diff
@@ -0,0 +1,260 @@
+    @Test(groups = {"python"}) // python environment is required to run tool
+    public void testNoResources() {
+        final File outputDir = createTempDir("score");
+        final String outputPrefix = String.format("%s/test", outputDir);
+        final ArgumentsBuilder argsBuilder = BASE_ARGUMENTS_BUILDER_SUPPLIER.get();
+        argsBuilder.add(LabeledVariantAnnotationsWalker.MODE_LONG_NAME, VariantType.SNP)
+                .addOutput(outputPrefix);
+        final String modelPrefix = new File(INPUT_FROM_TRAIN_EXPECTED_TEST_FILES_DIR,
+                "extract.nonAS.snpIndel.posUn.train.snp.posOnly.IF").toString();
+        final Function<ArgumentsBuilder, ArgumentsBuilder> addModelPrefix = ab ->
+                ADD_MODEL_PREFIX.apply(ab, modelPrefix);
+        addModelPrefix
+                .andThen(ExtractVariantAnnotationsIntegrationTest.ADD_NON_ALLELE_SPECIFIC_ANNOTATIONS)
+                .apply(argsBuilder);
+        runCommandLine(argsBuilder);
+        Assert.assertTrue(new File(outputPrefix + ScoreVariantAnnotations.ANNOTATIONS_HDF5_SUFFIX).exists());
+        Assert.assertTrue(new File(outputPrefix + ScoreVariantAnnotations.SCORES_HDF5_SUFFIX).exists());
+        Assert.assertTrue(new File(outputPrefix + ".vcf").exists());
+        Assert.assertTrue(new File(outputPrefix + ".vcf.idx").exists());
+    }
+
+    /**
+     * If no variants are present in the input in the specified region, we do not create the scores or annotations HDF5 files.
+     * This is because we cannot create HDF5 files with empty arrays/matrices.
+     */
+    @Test(groups = {"python"}) // python environment is required to run tool
+    public void testNoVariantsInInput() {
+        final File outputDir = createTempDir("score");
+        final String outputPrefix = String.format("%s/test", outputDir);
+        final ArgumentsBuilder argsBuilder = BASE_ARGUMENTS_BUILDER_SUPPLIER.get();
+        argsBuilder.add(LabeledVariantAnnotationsWalker.MODE_LONG_NAME, VariantType.SNP)
+                .add(StandardArgumentDefinitions.INTERVALS_LONG_NAME, "chr2") // the test input VCF does not have variants here
+                .addOutput(outputPrefix);
+        final String modelPrefix = new File(INPUT_FROM_TRAIN_EXPECTED_TEST_FILES_DIR,
+                "extract.nonAS.snpIndel.posUn.train.snp.posOnly.IF").toString();
+        final Function<ArgumentsBuilder, ArgumentsBuilder> addModelPrefix = ab ->
+                ADD_MODEL_PREFIX.apply(ab, modelPrefix);
+        addModelPrefix
+                .andThen(ExtractVariantAnnotationsIntegrationTest.ADD_NON_ALLELE_SPECIFIC_ANNOTATIONS)
+                .andThen(ExtractVariantAnnotationsIntegrationTest.ADD_SNP_MODE_AND_RESOURCES)
+                .apply(argsBuilder);
+        runCommandLine(argsBuilder);
+        Assert.assertFalse(new File(outputPrefix + ScoreVariantAnnotations.ANNOTATIONS_HDF5_SUFFIX).exists());
+        Assert.assertFalse(new File(outputPrefix + ScoreVariantAnnotations.SCORES_HDF5_SUFFIX).exists());
```

Comment:
Yes, unfortunately this might be a limitation of our HDF5 wrapper (https://github.com/broadinstitute/hdf5-java-bindings/blob/master/src/main/java/org/broadinstitute/hdf5/HDF5File.java) and perhaps even the Java HDF5 library itself. Our wrapper, at least, disallows writing of annotation matrices with no elements (ideally, we'd want to write a 0 x M matrix, where M is the number of features, so that we could at least encode M). I'm not sure if this is also true at the library level (or true for double arrays at either the wrapper or library level), but we could check.

In general, some Googling shows that the treatment of HDF5 null/empty/etc. arrays has evolved a bit over time and may differ across different versions/implementations (e.g., h5py). I think we'd want to match h5py/numpy behavior, if possible, since I would guess this is what most downstream code would use.

I'm not sure if `touch` will get around this without special treatment in downstream code, since an empty text file cannot really be treated the same as an HDF5 file with empty matrices/arrays. Perhaps optional outputs would be a better option (no pun intended)?

In any case, I'll file an issue and leave this to future work. Changing our HDF5 wrapper at all seems to add a lot of overhead...it hasn't really been updated in quite some time.

Reply:
Ah right, I'm not actually using the HDF5's downstream in the WDL so the empty file output didn't matter. I can update it to an optional output.